### DISCUSSION TITLE
Issues with Login Window Text Enable

### DISCUSSION THREAD

**Discussion by fnieves433** [usnistgov/macos_security#381] [General] [answered]
I keep getting a failed result with this setting. When enabled it installs properly but it cannot be detected. This is the result when the remediation script runs. 

Thu Mar 28 15:58:37 UTC 2024 system_settings_loginwindow_loginwindowtext_enable failed (Result: , Expected: "base64: vghpcybjb21wdxrlcibzexn0zw0gaxmgdghlihbyb3blcnr5ig9mierpz2l0ywwgrmvkzxjhbcbdcmvkaxqgvw5pb24uiel0iglzigzvcibhdxrob3jpemvkihvzzsbvbmx5libvc2vycyaoyxv0ag9yaxplzcbvcib1bmf1dghvcml6zwqpighhdmugbm8gzxhwbgljaxqgb3igaw1wbgljaxqgzxhwzwn0yxrpb24gb2ygchjpdmfjes4gqw55ig9yigfsbcb1c2vzig9mihroaxmgc3lzdgvtigfuzcbhbgwgzmlszxmgb24gdghpcybzexn0zw0gbwf5igjligludgvyy2vwdgvklcbtb25pdg9yzwqsihjly29yzgvklcbjb3bpzwqsigf1zgl0zwqsigluc3bly3rlzcwgyw5kigrpc2nsb3nlzcb0bybeawdpdgfsiezlzgvyywwgq3jlzgl0ifvuaw9uig1hbmfnzw1lbnqsigfuzcbsyxcgzw5mb3jjzw1lbnqgcgvyc29ubmvslcbhcyb3zwxsigfzigf1dghvcml6zwqgb2zmawnpywxzig9mig90agvyigfnzw5jawvzlcbib3roigrvbwvzdgljigfuzcbmb3jlawdulibcesb1c2luzyb0aglzihn5c3rlbswgdghlihvzzxigy29uc2vudhmgdg8gc3vjacbpbnrlcmnlchrpb24sig1vbml0b3jpbmcsihjly29yzgluzywgy29wewluzywgyxvkaxrpbmcsigluc3bly3rpb24gyw5kigrpc2nsb3n1cmugyxqgdghligrpc2nyzxrpb24gb2ygrglnaxrhbcbgzwrlcmfsienyzwrpdcbvbmlvbi4gvw5hdxrob3jpemvkig9yigltchjvcgvyihvzzsbvzib0aglzihn5c3rlbsbtyxkgcmvzdwx0igluigfkbwluaxn0cmf0axzligrpc2npcgxpbmfyesbhy3rpb24gyw5kignpdmlsigfuzcbjcmltaw5hbcbwzw5hbhrpzxmgqnkgy29udgludwluzyb0byb1c2ugdghpcybzexn0zw0gew91igluzgljyxrlihlvdxigyxdhcmvuzxnzig9migfuzcbjb25zzw50ihrvihrozxnlihrlcm1zigfuzcbjb25kaxrpb25zig9mihvzzs4gte9hie9gribjtu1frelbvevmwsbpzib5b3ugzg8gbm90igfncmvlihrvihrozsbjb25kaxrpb25zihn0yxrlzcbpbib0aglzihdhcm5pbmckckfuvevovelptiehifbmruftrsbsrufeienbukvgvuxmws4k")

**Comment by brodjieski** [comment]
Looking at the log entry, it's showing `(Result: , Expected:...` It appears that it's not finding a value when it's performing the check. Can you double-check that there is a profile deployed to the system that has the `LoginwindowText` key is set in the `com.apple.loginwindow` domain? You can use `sudo profiles -P -o stdout` to review all of the profiles applied to the system.

**Reply by fnieves433** [reply]
![Screenshot 2024-03-28 at 2 01 03 PM](https://github.com/usnistgov/macos_security/assets/62343270/29957fb3-a10d-487b-a633-6bab68fc04b0)

**Reply by brodjieski** [reply]
Interesting, it looks like it's there.  Do you happen to have another profile attempting to set that same key?

Do you get anything using this?
```
/usr/bin/osascript -l JavaScript << EOS
$.NSUserDefaults.alloc.initWithSuiteName('com.apple.loginwindow')\
.objectForKey('LoginwindowText').js
EOS
```

**Reply by fnieves433** [reply]
I get no result when running that command

XXX@lXXX ~ % /usr/bin/osascript -l JavaScript << EOS
$.NSUserDefaults.alloc.initWithSuiteName('com.apple.loginwindow')\
.objectForKey('LoginwindowText').js
EOS
XXXX@XXX


sorry I am trying to include a screenshot but I am having issues uploading.

**Reply by brodjieski** [reply, marked as answer]
We have seen strange behavior when there are special characters, particularly quotes, in the banner text.  I don't know for sure if this is what you are running into, but maybe try to deploy a profile with more simple text... just as a test.  Question.. with this current profile, do you see the text at the loginwindow screen?

**Reply by fnieves433** [reply]
removing the special characters !!! resolved the issue thank you!

**Comment by fnieves433** [comment]
It does apply the text correctly, 

Yahoo Mail: Search, Organize, Conquer 
 
  On Thu, Mar 28, 2024 at 5:07 PM, Dan ***@***.***> wrote:   


We have seen strange behavior when there are special characters, particularly quotes, in the banner text. I don't know for sure if this is what you are running into, but maybe try to deploy a profile with more simple text... just as a test. Question.. with this current profile, do you see the text at the loginwindow screen?
